### DISCUSSION TITLE
Avoid unnecessary API calls with SSR

### DISCUSSION THREAD

**Discussion by prateek3255** [vercel/next.js#22787] [Help]
I have a few pages which I want to be server-rendered, all of those pages make an API request to the same endpoint to render the initial state on the server but then they fetch some different data on the client-side using swr.

Right now these pages have their individual getServerSideProps which essentially fetches the same set of props for all these pages.

So I wanted to ask if there's a way that when navigating to one such page from another via client-side routing to avoid fetching the same props in the getServerSideProps of another page because we already have fetched that data in the getServerSideProps of the current page

**Comment by ErnestBrandi** [comment]
Hi, no I don't think this is possible neither wanted as data returned from an API is usually not consistent.
But why don't you use SWR to fetch this data also ? Then, with the proper set of options, data is shared across web pages and only revalidated upon certain actions / time.

**Reply by prateek3255** [reply]
Yes, I know fetching this data on the client with SWR would solve my issue, but I want some part of all these pages to be server-rendered for SEO purposes.

**Reply by ErnestBrandi** [reply]
I believe client side rendering is less and less an issue with SEO as bots render the pages anyway.
Also you could store that API response in some way (with a timestamp for revalidation) and use this instead of making another API call.
